### PR TITLE
[JFG-944] Migrate jaas test to java configuraion

### PR DESCRIPTION
**Pull request by oskliarov**
- Migrate jaas test from  JFG-933 (https://github.com/griddynamics/jagger/pull/638) to java configuration
- add tests for /executions

### CODE REVIEW

File: test/jaas-test/src/main/java/com/griddynamics/jagger/test/jaas/JaasScenario.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                getTestWithResponseCode200("t_JaaS_GET_Exact_Session_Test_SR_eq._1.0", queryProvider.GET_TestNames(), TestResponseContentValidator.class)
+        ).build();
+
+        JParallelTestsGroup tg_JaaS_GET_Metrics = getSingleTestWithResponseCode200("t_JaaS_GET_Metrics_List_SR_eq._1.0",
+                queryProvider.GET_TestMetrics(), MetricsListResponseContentValidator.class);
+
+        JParallelTestsGroup tg_JaaS_GET_Metric_Summaries = getSingleTestWithResponseCode200("t_JaaS_GET_Metric_Summary_List_SR_eq",
+                queryProvider.GET_MetricSummary(), SummaryListResponseContentValidator.class);
+
+        JParallelTestsGroup td_JaaS_MetricPlotList = getSingleTestWithResponseCode200("t_JaaS_GET_Metric_Plot_List_SR_eq._1.0",
```
Comment (dlatnikov) [inline test/jaas-test/src/main/java/com/griddynamics/jagger/test/jaas/JaasScenario.java]: td_ => tg_

File: test/jaas-test/src/main/java/com/griddynamics/jagger/test/jaas/invoker/ExecutionManipulateInvoker.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import com.griddynamics.jagger.invoker.InvocationException;
+import com.griddynamics.jagger.invoker.v2.JHttpEndpoint;
+import com.griddynamics.jagger.invoker.v2.JHttpQuery;
+import com.griddynamics.jagger.invoker.v2.JHttpResponse;
+import com.griddynamics.jagger.test.jaas.util.TestContext;
+import com.griddynamics.jagger.test.jaas.util.entity.ExecutionEntity;
+import org.springframework.http.HttpMethod;
+
+public class ExecutionManipulateInvoker extends InvokerWithoutStatusCodeValidation {
```
Comment (dlatnikov) [inline test/jaas-test/src/main/java/com/griddynamics/jagger/test/jaas/invoker/ExecutionManipulateInvoker.java]: please add javadoc for every invoker, validator. it should be clear what is the goal of every class

File: test/jaas-test/src/main/java/com/griddynamics/jagger/test/jaas/validator/metrics/MetricsListResponseContentValidator.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        //TODO: Refactor when JFG-943  is ready.
+        assertTrue("Actual list is not the same as expected set.", compareIgnoringTestDto(expectedEntities, actualEntities));
+
+        return true;
+    }
+
+    private String getSessionIdFromQuery(JHttpQuery query) {
+        // ${jaas.rest.root}/sessions/{sessionId}/tests/{testName}/metrics => ${jaas.rest.root} + sessions + {sessionId} + tests + {testName} + metrics
+        // TODO : re-factor
```
Comment (dlatnikov) [inline test/jaas-test/src/main/java/com/griddynamics/jagger/test/jaas/validator/metrics/MetricsListResponseContentValidator.java]: good style to add JFG-xxxx to every todo

File: test/jaas-test/src/main/resources/suite/invokers.conf.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns="http://www.springframework.org/schema/beans"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd"
+       default-lazy-init="true">
+
+    <!--This xml file will contain bean declaration of custom invokers-->
```
Comment (dlatnikov) [inline test/jaas-test/src/main/resources/suite/invokers.conf.xml]: please mark as temporary workaround with todo

File: test/jaas-test/src/main/java/com/griddynamics/jagger/test/jaas/provider/QueryProvider.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    public Iterable GET_ExId() {
+        return () -> TestContext.getCreatedExecutionIds().stream()
+                .map(id -> new JHttpQuery<String>()
+                        .get().responseBodyType(ExecutionEntity.class)
+                        .path(executions_uri + "/" + id))
+                .iterator();
+    }
+
+    public Iterable GET_NonNumeric_ExId() {
+        return () -> Stream.of("/abvgdeyka", "/ABVGD")
```
Comment (SokolAndrey) [inline test/jaas-test/src/main/java/com/griddynamics/jagger/test/jaas/provider/QueryProvider.java]: 👍

File: test/jaas-test/src/main/java/com/griddynamics/jagger/test/jaas/JaasScenario.java [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+import com.griddynamics.jagger.engine.e1.collector.NotNullResponseValidator;
+import com.griddynamics.jagger.engine.e1.collector.ResponseValidator;
+import com.griddynamics.jagger.invoker.v2.JHttpEndpoint;
+import com.griddynamics.jagger.test.jaas.invoker.ExecutionManipulateInvoker;
+import com.griddynamics.jagger.test.jaas.invoker.InvokerWithoutStatusCodeValidation;
+import com.griddynamics.jagger.test.jaas.listener.LoadScenarioConfigListener;
+import com.griddynamics.jagger.test.jaas.provider.QueryProvider;
+import com.griddynamics.jagger.test.jaas.util.TestContext;
+import com.griddynamics.jagger.test.jaas.validator.*;
```
Comment (SokolAndrey) [inline test/jaas-test/src/main/java/com/griddynamics/jagger/test/jaas/JaasScenario.java]: Not good

File: test/jaas-test/src/main/java/com/griddynamics/jagger/test/jaas/invoker/InvokerWithoutStatusCodeValidation.java [hunk 7]
```diff
@@ -7,6 +7,10 @@
 import java.util.HashMap;
 import java.util.Map;
 
+/**
+ * Invoker without checking result codes.
+ * I.e. if request returns 4xx or 5xx this invoker doesn't throw exception unlike DefaultHttpInvoker
+ */
```
Comment (dlatnikov) [inline test/jaas-test/src/main/java/com/griddynamics/jagger/test/jaas/invoker/InvokerWithoutStatusCodeValidation.java]: such behavior of the Default invoker is a bug
https://issues.griddynamics.net/browse/JFG-1059

**Pull request merged**